### PR TITLE
Refactor remaining end to end tests

### PR DESCRIPTION
**Pull request by svacas**

### CODE REVIEW

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            </batch:process-records>
+            <batch:on-complete>
+                <logger message="something" />
+            </batch:on-complete>
+        </batch:job>
+
+    </flow>
+
+    <sub-flow name="upperCasePipe">
+        <set-payload value="#[mel:payload + '|' + payload.toUpperCase()]">
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml:37]: maybe its a good case to analyze why the expression is not migrated?

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/ftp/ftp/output/src/main/mule/mule-config.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:compatibility="http://www.mulesoft.org/schema/mule/compatibility" xmlns:ftp="http://www.mulesoft.org/schema/mule/ftp" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/ftp http://www.mulesoft.org/schema/mule/ftp/current/mule-ftp.xsd http://www.mulesoft.org/schema/mule/compatibility http://www.mulesoft.org/schema/mule/compatibility/current/mule-compatibility.xsd http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+
+    <ftp:config name="ftpSource">
+        <ftp:connection host="localhost" port="${ftpSourcePort}" username="admin" password="admin" passive="true" transferMode="BINARY">
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/ftp/ftp/output/src/main/mule/mule-config.xml:5]: the previous test is throwing errors saying that FTP is not supported, but this one does not. One of them might be wrong

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+
+    <flow name="withsampledatafilesFlow">
+        <set-payload value="#[payload]" mimeType="application/xml" />
+
+        <ee:transform doc:name="Transform Message">
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml:7]: Not sure if this is a valid test, by the directory name this should be reported as unsupported, but the error here is different. Maybe is a case we need to handle

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/HttpListener.dwl [hunk 4]
```diff
@@ -0,0 +1,29 @@
+%dw 2.0
+
+/**
+ * Emulates the response headers building logic of the Mule 3.x HTTP Connector.
+ */
+fun httpListenerResponseHeaders(vars: {}) = do {
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/HttpListener.dwl:6]: I think we are not comparing these files in other tests, aren't we? Do we want to add them to all the tests? or remove from this one to keep consistency?

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/attributes2inboundProperties.dwl [hunk 5]
```diff
@@ -0,0 +1,27 @@
+%dw 2.0
+output application/java
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/attributes2inboundProperties.dwl:2]: I think we are not comparing these files in other tests, aren't we? Do we want to add them to all the tests? or remove from this one to keep consistency?

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/policy/policy_simple/output/report/report.json [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  "numberOfDWTransformations": 0,
+  "numberOfDWTransformationsMigrated": 0,
+  "numberOfDWTransformationLines": 0,
+  "numberOfDWTransformationLinesMigrated": 0,
+  "detailedMessages": [
+    {
+      "level": "INFO",
+      "key": "basicStructure.pointcutMigrationStepResolvedByRuntime",
+      "component": "pointcut",
+      "lineNumber": -1,
```
Comment (jesica-fera) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/policy/policy_simple/output/report/report.json:42]: line number is -1, this comment is deleted on the final migrated file

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml (hunk 1 above)
Comment (svacas) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml:37]: I'll include this as part of the task to analyze mel expressions

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml (hunk 3 above)
Comment (svacas) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml:7]: It's an unsupported attribute `metadata:id`, the error says more or less that:
`Custom types defined in Studio 6 are not migrated to Studio 7`

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/HttpListener.dwl (hunk 4 above)
Comment (svacas) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/oauth2/tweetbook-oauth2-provider/output/src/main/resources/migration/HttpListener.dwl:6]: I've added them in order to have at lease one test that verifies that those files are generated, somewhat like the html report test, but maybe it should be a separate test for that purpose only

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/policy/policy_simple/output/report/report.json (hunk 6 above)
Comment (jesica-fera) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/policy/policy_simple/output/report/report.json:42]: We can tackle this as a separate task

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml (hunk 3 above)
Comment (jesica-fera) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/withsampledatafiles/output/src/main/mule/withsampledatafiles.xml:7]: Ok, I was just confused by the purpose of the test. I think I understand now

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml (hunk 1 above)
Comment (jesica-fera) [reply mule-migration-tool-e2e-tests/src/test/resources/e2e/batch/batch1/output/src/main/mule/mule-config.xml:37]: Sounds good

**Review by jesica-fera — APPROVED** [review]

**Pull request merged**